assets/queries/openAPI/string_schema_with_broad_pattern/query.rego

```diff
@@ -0,0 +1,70 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	[path, value] := walk(doc)
+	schema_kind = openapi_lib.get_schema(value)
+	info := openapi_lib.is_operation(path)
+	openapi_lib.content_allowed(info.operation, info.code)
+	has_broad_schema(value, schema_kind, doc)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.%s", [openapi_lib.concat_path(path), schema_kind]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "String schema has 'pattern' restricted",
+		"keyActualValue": "String schema does not have 'pattern' restricted",
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	[path, value] := walk(doc)
+	schema_kind = openapi_lib.get_schema(value)
+	openapi_lib.is_operation(path) == {}
+	has_broad_schema(value, schema_kind, doc)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.%s", [openapi_lib.concat_path(path), schema_kind]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "String schema has 'pattern' restricted",
+		"keyActualValue": "String schema does not have 'pattern' restricted",
+	}
+}
```

Comment:
I think this can be a single rule as following:
```suggestion
CxPolicy[result] {
	doc := input.document[i]
	openapi_lib.check_openapi(doc) != "undefined"

	[path, value] := walk(doc)
  pattern := value.pattern
	broad_schema(pattern) # Also needs to update broad_schema to receive pattern

	result := {
		"documentId": doc.id,
		"searchKey": sprintf("%s.pattern", [openapi_lib.concat_path(path)]),
		"issueType": "IncorrectValue",
		"keyExpectedValue": "String schema has 'pattern' restricted",
		"keyActualValue": "String schema does not have 'pattern' restricted",
	}
}
```